src/configuration/converters.ts

```diff
@@ -0,0 +1,120 @@
+import {
+  FareProductTypeConfigSettings,
+  FareProductTypeConfig,
+  OfferEndpoint,
+  ProductSelectionMode,
+  TimeSelectionMode,
+  TravellerSelectionMode,
+  ZoneSelectionMode,
+  FareProductType,
+} from '@atb/screens/Ticketing/FareContracts/utils';
+import {LanguageAndTextType} from '@atb/translations/types';
+import {isArray} from 'lodash';
+
+export function mapToFareProductTypeConfigs(
+  config: any,
+): FareProductTypeConfig[] | undefined {
+  if (!isArray(config)) return;
+
+  return config
+    .map((val) => mapToFareProductTypeConfig(val))
+    .filter(Boolean) as FareProductTypeConfig[];
+}
+
+function mapToFareProductTypeConfig(
+  config: any,
+): FareProductTypeConfig | undefined {
+  const fields = [
+    'type',
+    'name',
+    'transportModes',
+    'description',
+    'configuration',
+  ];
+  if (!fields.every((f) => f in config)) return;
+
+  if (!isArray(config.name) || !isArray(config.description)) return;
+
+  let name = mapLanguageAndTextType(config.name);
+  let description = mapLanguageAndTextType(config.description);
+
+  if (!name || !description) return;
+
+  const fcType = mapToStringAlternatives<FareProductType>(config.type, [
+    'single',
+    'period',
+    'hour24',
+    'night',
+  ]);
+  if (!fcType) return;
+
+  const configuration = mapToFareProductConfigSettings(config.configuration);
+  if (!configuration) return;
+
+  return {
+    type: fcType,
+    name,
+    description,
+    transportModes: config.transportModes,
```

Comment:
Should probably validate `transportModes` too. Maybe check that it is an array of strings.

Reply:
@rosvik `fare-product-type-config.json` is based on the actual values on the remote-config branch.
Do you want to record each wrong value or one bugsnag report to the entire configuration?

Reply:
> `fare-product-type-config.json` is based on the actual values on the remote-config branch.

Yes, it's probably fine. I was thinking if we want to show every product if the device doesn't have access to Firestore, or if we should hide periodic tickets for example. I don't know, but shouldn't be a blocker for this PR 😊

> Do you want to record each wrong value or one bugsnag report to the entire configuration?

I'm not sure how it should work, but I guess it would be nice if we had an indication on what part is wrong.

Reply:
@rosvik I made fixes to include detailed bugsnag descriptions if something is wrong. Please take a look!